workflows/diagnostics/fv3net/diagnostics/offline/compute.py

```diff
@@ -294,14 +291,43 @@ def main(args):
         f_out.write(f_in.read())
     batches = config.load_batches(model_variables)
 
-    predict_function = _get_predict_function(model, model_variables, grid)
-    batches = loaders.Map(predict_function, batches)
+    transforms = [_get_predict_function(model, model_variables)]
+
+    prediction_resolution = res_from_string(config.res)
```

Comment:
Could you move this to a couple functions in a separate coarsening module? 
ex.
```
if needs_to_coarsen(...) is True:
    transforms.append( get_coarsening_transform(...))

```